[ITLimJiaWei] iP

ITLimJiaWei opened the pull request:
## Hoshi

"Experience is the name everyone gives to their mistakes." -  Oscar Wilde [(source)](https://www.goodreads.com/quotes/7701986-experience-is-the-name-every-one-gives-to-their-mistakes)



- Text-Based
- Easy to use
- Feature rich

Coded with **OOP** in mind! :smile:
This is part of `Hoshi.java`
``` java
private Storage storage;
    private TaskList taskList;
    private Ui ui;
    private Parser parser;

```

1. Download JAR file
2. Run JAR file

Features
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [x] Listing all tasks
- [ ] Graphical User Interface(coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ITLimJiaWei committed:
Add rename, greet and exit functionality

ITLimJiaWei committed:
Add acknowledgements to ASCII art site

ITLimJiaWei committed:
Add echo feature

ITLimJiaWei committed:
Add add and list features

ITLimJiaWei committed:
Add task class

ITLimJiaWei committed:
Add support for mark as done

ITLimJiaWei committed:
Add deadline, event and todo classes

ITLimJiaWei committed:
Update task to be abstract and added toString method

ITLimJiaWei committed:
Add support for todos, events, deadlines

ITLimJiaWei committed:
Add simple error handling for invalid input

ITLimJiaWei committed:
Add javadocs

ITLimJiaWei committed:
Add HoshiException class and support for error handling

ITLimJiaWei committed:
Add support for delete and more error handling

ITLimJiaWei committed:
fix mark/unmark error handling

ITLimJiaWei committed:
Add automated text ui testing

ITLimJiaWei committed:
Add error handling for when user inputs just mark, unmark, delete or add

ITLimJiaWei committed:
Fix switch case indentation and starton README

ITLimJiaWei committed:
Remove deprecated code

ITLimJiaWei committed:
Revert "Remove deprecated code"

This reverts commit 777e0833d93ba42713879c6d869ebd2da1d0e01c.

ITLimJiaWei committed:
Add support for save feature and improved JavaDocs

ITLimJiaWei committed:
Add support for deadline and event datetimes

ITLimJiaWei committed:
Add Ui class

ITLimJiaWei committed:
Add storage class

ITLimJiaWei committed:
Update Storage and Ui Javadocs

ITLimJiaWei committed:
Add Parser class and more Ui functionality

ITLimJiaWei committed:
Add TaskList

ITLimJiaWei committed:
Add support for MoreOOP and Fix datetime toString

ITLimJiaWei committed:
Added more support for MoreOOP

ITLimJiaWei committed:
Add packaging and fix delete bug

ITLimJiaWei committed:
fix issue where tasks are only saved on program termination

ITLimJiaWei committed:
Remove repetitive code for saving task

ITLimJiaWei committed:
Remove irrelevant code

ITLimJiaWei committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

ITLimJiaWei committed:
Add several JUnit tests for Parser class

ITLimJiaWei committed:
Remove deprecated methods after packaging

ITLimJiaWei committed:
Add support for JUnit testing for Parser class

ITLimJiaWei committed:
Add Javadocs

ITLimJiaWei committed:
Fix several coding standard issues

ITLimJiaWei committed:
Add support for Find feature

ITLimJiaWei committed:
Merge branch 'branch-A-CodingStandard' into master

# Conflicts:
#	src/main/java/hoshi/Hoshi.java
#	src/main/java/hoshi/task/TaskList.java
#	src/main/java/hoshi/utils/Parser.java

ITLimJiaWei committed:
Merge branch 'branch-Level-9' into master

# Conflicts:
#	src/main/java/hoshi/ui/Ui.java
#	src/main/java/hoshi/utils/Parser.java

KennethTeo2002 commented on .gitignore:
```diff
@@ -15,3 +15,4 @@ bin/
 
 /text-ui-test/ACTUAL.TXT
```
Some of your files using absolute path while others are using relative path. It would be good to standardize your file to either.

KennethTeo2002 commented on README.md:
```diff
@@ -13,12 +13,12 @@ Prerequisites: JDK 17, update Intellij to the most recent version.
    |  _ \ _   _| | _____ 
    | | | | | | | |/ / _ \
    | |_| | |_| |   <  __/
    |____/ \__,_|_|\_\___|
```
This should preferably be changed to the actual output users will see when compiling your project.

KennethTeo2002 commented on docs/README.md:
```diff
@@ -1,9 +1,29 @@
 
-// Update the title above to match the actual product name
 
 // Product screenshot goes here
```
Do you intend for these comments to be shown on the document? If not, you could use html commenting 
```<!-- comment -->```

KennethTeo2002 commented on docs/README.md:
```diff
@@ -27,4 +47,9 @@ expected output
+// Feature details
+
+
+## Acknowledgements
```
Good job for acknowledging external sources.

KennethTeo2002 commented on src/main/java/hoshi/Hoshi.java:
```diff
@@ -0,0 +1,58 @@
+        ui = new Ui();
+        storage = new Storage(filePath);
+        taskList = new TaskList();
+        parser= new Parser();
```
Missing spacing.

KennethTeo2002 commented on src/main/java/hoshi/task/Deadline.java:
```diff
@@ -0,0 +1,61 @@
+ * of the deadline, whether the task has been completed or not and the end time.</p>
+ */
+
+public class Deadline extends Task{
```
Missing spacing. Applicable to other files as well.

KennethTeo2002 commented on src/main/java/hoshi/task/Deadline.java:
```diff
@@ -0,0 +1,61 @@
+     * @return description endTime which indicates when the Deadline is due by.
+     */
+    public LocalDate getEndTime() {
+        return endTime;
```
Some places you use this.varName, would be better to remain consistent. Applicable to other files as well.

KennethTeo2002 commented on src/main/java/hoshi/task/Event.java:
```diff
@@ -0,0 +1,74 @@
+ * <p>This class inherits from the Task class It maintains a description
+ * of the event, whether the event has been completed or not and its start and end time</p>
+ */
+public class Event extends Task{
```
Missing spacing.

KennethTeo2002 commented on src/main/java/hoshi/task/Deadline.java:
```diff
@@ -0,0 +1,61 @@
+
+public class Deadline extends Task{
+
+    protected LocalDate endTime;
```
Could this be private? Applicable to other files as well.

KennethTeo2002 commented on src/main/java/hoshi/task/TaskList.java:
```diff
@@ -0,0 +1,58 @@
+
+public class TaskList {
+
+    private ArrayList<Task> taskList;
```
Collection of tasks should be named as a plural.

KennethTeo2002 commented on src/test/java/hoshi/utils/ParserTest.java:
```diff
@@ -0,0 +1,250 @@
+     * Tests the success case of handleMark function
+     */
+    @Test
+    public void handleMarkTest_markCompleted_success(){
```
Missing spacings for all methods.

KennethTeo2002 reviewed (commented):
LGTM.
A few coding standard violations here and there, but overall, a really good job👍

ITLimJiaWei committed:
Incorporate feedback from 1st PR review

ITLimJiaWei commented:
Thank you for the comments, will incorporate the feedback 👍 👍

ITLimJiaWei committed:
Remove redundant code

ITLimJiaWei committed:
Add support for CheckStyle

ITLimJiaWei committed:
Add barebones support for GUI and fixed mocking issue in ParserTest

ITLimJiaWei committed:
Add partial support for GUI (list, bye, mark/unmark features only for now)

ossaris56 commented on src/main/java/hoshi/task/Deadline.java:
```diff
@@ -0,0 +1,62 @@
+
+public class Deadline extends Task {
+
+    private LocalDate endTime;
```
I think its a bit misleading to call it endTime as the type is dealing with Dates not DateTimes. EndDate would be better naming wise in my opinion.

ossaris56 commented on src/main/java/hoshi/task/Event.java:
```diff
@@ -0,0 +1,75 @@
+ */
+public class Event extends Task {
+
+    private LocalDate startTime;
+
+    private LocalDate endTime;
```
Similar to above, they should be named startDate and endDate.

ossaris56 reviewed (commented):
LGTM! Just some minor naming violations. Overall your code is easy to read and well structured!

ITLimJiaWei committed:
Improve GUI

ITLimJiaWei committed:
Fix bye command and add 2s delay before terminating

ITLimJiaWei committed:
Incorporate PR feedback and improve css

ITLimJiaWei committed:
Add support for JavaFX GUI

ITLimJiaWei committed:
Update README

ITLimJiaWei committed:
Update README

ITLimJiaWei committed:
Update README

ITLimJiaWei committed:
Improve code quality

ITLimJiaWei committed:
Improve code standard for HandleAdd in Parser class

ITLimJiaWei committed:
Fix JUnit test bug caused by deprecation of CLI

ITLimJiaWei committed:
Add support for Java Assertions feature

Parser and Storage classes: add Java Assertions

With Java assertions, the developer wille be able to more quickly
find issues or bugs within the code if any do occur.

Java assertions are added at certain points to ensure fields such
as input, description and markIndex are not null or empty(if
applicable).

These assertions are placed at these points as these variables are
more prone causing bugs related to null values.

ITLimJiaWei committed:
Update classes to adhere to SLAP principle

Hoshi and Parser Classes: Adhere to SLAP

These changes are made to improve code readability which would lead
to easier maintenance and faster debugging.

Method calls are separated from usual lines of code at appropiate
places by abstracting code into smaller functions. In addition to
this, Parser logic has been separated into Command classes for
better readability.

In order to ensure methods have a single layer of abstraction.

ITLimJiaWei committed:
Update storage class to adhere to SLAP

ITLimJiaWei committed:
Merge pull request #1 from ITLimJiaWei/branch-A-CodeQuality

A-CodeQuality PR

ITLimJiaWei committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/hoshi/utils/Parser.java
#	src/main/java/hoshi/utils/Storage.java

ITLimJiaWei committed:
Fix merge conflicts when merging master to branch-A-Assertions

ITLimJiaWei committed:
Merge pull request #2 from ITLimJiaWei/branch-A-Assertions

branch-A-Assertions PR

ITLimJiaWei committed:
Add support for Github Actions CI

ITLimJiaWei committed:
Add gradle.yml file for Github Actions CI

ITLimJiaWei committed:
Fix checkstyle bug causing CI failure

ITLimJiaWei committed:
Merge pull request #3 from ITLimJiaWei/branch-A-IC

Branch-A-IC PR

ITLimJiaWei committed:
Redo several JUnit tests to support command classes

ITLimJiaWei committed:
Redo JUnit tests for command classes

ITLimJiaWei committed:
Remove IO redirection tests from CI

ITLimJiaWei committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/test/java/hoshi/command/AddCommandTest.java
#	src/test/java/hoshi/utils/ParserTest.java

ITLimJiaWei committed:
Add support for C-Help

ITLimJiaWei committed:
Remove redundant variables

ITLimJiaWei committed:
Fix checkstyle

ITLimJiaWei committed:
Merge pull request #4 from ITLimJiaWei/branch-BCD-Extension

Branch-BCD-extension PR

ITLimJiaWei committed:
Fix empty input bug

ITLimJiaWei committed:
Merge remote-tracking branch 'origin/master'

ITLimJiaWei committed:
Replace magic numbers with enumerable

ITLimJiaWei committed:
Improve code quality, add more command classes and update README

ITLimJiaWei committed:
Remove deprecated code for command line version

ITLimJiaWei committed:
Update README.md

ITLimJiaWei committed:
Improve code quality and fix storage loading bug

ITLimJiaWei committed:
Improve code quality

ITLimJiaWei committed:
Improve code quality

ITLimJiaWei committed:
Merge remote-tracking branch 'origin/master'

ITLimJiaWei committed:
Update README

ITLimJiaWei committed:
Update README and add Ui screenshot

ITLimJiaWei committed:
Fix JUnit tests

ITLimJiaWei committed:
Merge pull request #6 from ITLimJiaWei/branch-A-UserGuide

PR branch-A-UserGuide

ITLimJiaWei committed:
Update README.md

ITLimJiaWei committed:
Improve code standard

ITLimJiaWei committed:
Update acknowledgements

ITLimJiaWei committed:
Fix bug where description did not allow spaces

ITLimJiaWei committed:
Update README.md

ITLimJiaWei committed:
Improve code quality

ITLimJiaWei committed:
Fix checkstyle errors

ITLimJiaWei committed:
Improve code quality

ITLimJiaWei committed:
Add error validation for deadline/event edges cases

ITLimJiaWei committed:
Add additional unit tests for edge cases

ITLimJiaWei committed:
Add additional unit tests and add more error validation

ITLimJiaWei committed:
Fix data directory bug

ITLimJiaWei committed:
Add error handling to find command

ITLimJiaWei committed:
Fix deletecommand junit test

ITLimJiaWei committed:
Fix several input bugs and update README.md

ITLimJiaWei committed:
Fix help window typo

ITLimJiaWei committed:
Add remaining JavaDocs

ITLimJiaWei committed:
Update user guide

ITLimJiaWei committed:
Improve code quality

ITLimJiaWei committed:
Update comments and README

ITLimJiaWei committed:
Update user guide

ITLimJiaWei committed:
Update README

ITLimJiaWei committed:
Update README

ITLimJiaWei committed:
Fix int overflow bug
